src/os/newstore/NewStore.cc

```diff
@@ -3331,12 +3385,19 @@ int NewStore::_do_write_all_overlays(TransContext *txc,
     map<uint64_t,overlay_t>::iterator prev = p, next = p;
     ++next;
     while (next != o->onode.overlay_map.end()) {
-      if (prev->first + prev->second.length == next->first) {
+      if (prev->first + prev->second.length == next->first &&
+	  next->first < frag_first + o->onode.frag_size) {
```

Comment:
is that possible one overlay hit more than one fragment?
which is:
next->first < frag_first + o->onode.frag_size && next->first + next->second.length > frag_first + o->onode.frag_size

we need to chop split the next overlay key into two wal_op and handle the op->overlay clearly(seems txc->overlay will be better as only both op success we can delete the overlay)